app/src/main/java/in/co/rajkumaar/amritarepo/opac/OPACAdapter.java

```diff
@@ -58,28 +60,71 @@ public View getView(int position, @Nullable final View convertView, @NonNull Vie
         listItemView.setOnClickListener(new View.OnClickListener() {
             @Override
             public void onClick(View v) {
-                switch (current.getName()){
-                    case "Search Books" :
-                        context.startActivity(new Intent(context,OPACSearchActivity.class));
+                switch (current.getName()) {
+                    case "Search Books":
+                        context.startActivity(new Intent(context, OPACSearchActivity.class));
                         break;
                     case "E-Resources":
-                        context.startActivity(new Intent(context,EResourcesActivity.class));
+                        context.startActivity(new Intent(context, EResourcesActivity.class));
                         break;
                     case "Useful Sites":
-                        context.startActivity(new Intent(context,EResourcesActivity.class).putExtra("useful-sites",true));
+                        context.startActivity(new Intent(context, EResourcesActivity.class).putExtra("useful-sites", true));
                         break;
                     case "Account":
                         Dialog dj = new Dialog(context);
                         dj.setContentView(R.layout.library_patron_dialog);
                         final EditText username = dj.findViewById(R.id.id);
                         final EditText password = dj.findViewById(R.id.password);
+                        final CheckBox remember = dj.findViewById(R.id.remember_me);
+
+                        final SharedPreferences pref = context.getSharedPreferences("user", Context.MODE_PRIVATE);
```

Comment:
Why are you retrieving the normal shared prefs here? Also as I mentioned already, avoid initialising the enc to null initially and then reassigning it.
Instead you can go with final Encryption enc = new Encryption() inside the try block itself. Will that cause any harm? This way you don’t have to have two variables

Reply:
retrieving shared prefs to get the editor on button click. Again, need the Encryption object inside button click, so won't have visibility